test/Faker/Provider/ar_EG/CompanyTest.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Faker\Test\Provider\ar_EG;
+
+use Faker\Calculator\Luhn;
+use Faker\Provider\ar_EG\Company;
+use Faker\Test\TestCase;
+
+/**
+ * @group legacy
+ */
+final class CompanyTest extends TestCase
+{
+    public function testCompanyPrefix()
+    {
+        $companyPrefix = $this->faker->companyPrefix();
+        self::assertIsString($companyPrefix);
+    }
+
+    public function testCatchPhrase()
+    {
+        $catchPhrase = $this->faker->catchPhrase();
+        self::assertIsString($catchPhrase);
+    }
+
+    public function testCompanyTaxIdNumberIsValid()
+    {
+        $companyTaxIdNumber = $this->faker->companyTaxIdNumber;
+        self::assertMatchesRegularExpression('/\d{9}$/', $companyTaxIdNumber);
+        self::assertTrue(Luhn::isValid($companyTaxIdNumber));
+    }
+
+    public function testCompanyTradeRegisterNumberIsValid()
+    {
+        $companyTradeRegisterNumber = $this->faker->companyTradeRegisterNumber;
```

Comment:
Please use method invocations to get a random value, same for line 28.

Please test for a specific value, our test's Faker instance will always output the same value because it's seeded.